packages/verified-fetch/src/verified-fetch.ts

```diff
@@ -259,7 +252,27 @@ export class VerifiedFetch {
       this.log.error('error walking path %s', path, err)
       return badGatewayResponse(resource, 'Error walking path')
     }
-    const block = terminalElement?.node ?? await this.helia.blockstore.get(cid, options)
+  }
+
+  private async handleDagCbor ({ resource, cid, path, accept, options }: FetchHandlerFunctionArg): Promise<Response> {
+    this.log.trace('fetching %c/%s', cid, path)
+    let terminalElement: ObjectNode
+
+    // need to walk path, if it exists, to get the terminal element
+    const pathDetails = await this.handlePathWalking({ cid, path, resource, options })
+    if (pathDetails instanceof Response) {
+      return pathDetails
+    }
+    const ipfsRoots = pathDetails.ipfsRoots
+    if (isObjectNode(pathDetails.terminalElement)) {
+      terminalElement = pathDetails.terminalElement
+    } else {
+      // this should never happen, but if it does, we should log it and return notSupportedResponse
+      this.log.error('terminal element is not a dag-cbor node')
+      return notSupportedResponse(resource, 'Terminal element is not a dag-cbor node')
```

Comment:
not sure if this is the right thing to do here, or if we want to allow walking the path to fail and still query `helia.blockstore.get(cid, options)`